src/MSGEQ7.hpp

```diff
@@ -220,3 +214,21 @@ getVolume(void){
 	// return the average of all channels
 	return vol / 7;
 }
+
+template <MSGEQ7_data_t smooth, uint8_t resetPin, uint8_t strobePin, uint8_t firstAnalogPin, uint8_t ...analogPins>
+MSGEQ7_data_t CMSGEQ7<smooth, resetPin, strobePin, firstAnalogPin, analogPins ...>::
+getAnalogReadPin(uint8_t pin) {
+	int read = analogRead(pin);
+	#if defined(MSGEQ7_10BIT) || defined(MSGEQ7_12BIT)
+		// Return the value as int.
+		return read;
+	#else
+		// Some systems, esp8266, have range from 0 - 1024 so there is a max range of 1023.
+		uint16_t max10BitVal = 1023;
```

Comment:
I am not sure, does this give compiler warnings, because you compare signed vs unsigned? Maybe a #define should be used here? I am not sure, just want to think about it...